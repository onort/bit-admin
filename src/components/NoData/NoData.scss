@import "../../styles/main.scss";

.container {
  padding: $spacer-lg $spacer-sm;
}

.message {
  color: $text-color-muted;
  font-style: italic;
}
