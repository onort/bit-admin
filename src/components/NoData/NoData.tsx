import React from "react"
import cx from "classnames"

import styles from "./NoData.scss"

interface Props {
  className?: string
  message?: string
}

const NoData: React.SFC<Props> = props => {
  const className = cx(styles.container, props.className)
  return (
    <div className={className}>
      <p className={styles.message}>
        {props.message ? props.message : "No data to show."}
      </p>
    </div>
  )
}

export default NoData
