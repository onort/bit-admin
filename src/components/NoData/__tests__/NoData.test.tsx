import React from "react"
import { shallow } from "enzyme"

import NoData from "../"

const testMessage = "No data message test."
const testClassName = "testClassName"

describe("<NoData />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<NoData />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should render message", () => {
    const wrapper = shallow(<NoData message={testMessage} />)
    expect(wrapper.find(".message").text()).toBe(testMessage)
  })

  it("should have custom classname", () => {
    const wrapper = shallow(<NoData className={testClassName} />)
    expect(wrapper.find(".container").hasClass(testClassName)).toBe(true)
  })
})
