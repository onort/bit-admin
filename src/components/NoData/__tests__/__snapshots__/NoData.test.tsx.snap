// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoData /> should match snapshot 1`] = `
<div
  className="container"
>
  <p
    className="message"
  >
    No data to show.
  </p>
</div>
`;

exports[`<NoData /> should match snapshot 2`] = `"<div class=\\"container\\"><p class=\\"message\\">No data to show.</p></div>"`;
